routes/v1.js

```diff
@@ -1040,18 +1063,19 @@ api.declare({
   // Construct status object
   var status = task.status();
 
-  // Publish message about last run, if it was canceled
+  // If the last run was canceled, resolve dependencies and publish message
   if (run.state === 'exception' && run.reasonResolved === 'canceled') {
```

Comment:
spitballing, you could do a second level property resolution
a = { CANCELLED: {v: 'canceled'}}
then in your code, you could do
a.CANCELLED.v
and it would throw if you were to do
a.INVALIDENUM.v.

Ugly though...